airbyte-integrations/connectors/source-mysql/src/main/kotlin/io/airbyte/integrations/source/mysql/MysqlSourceOperations.kt

```diff
@@ -261,11 +265,33 @@ class MysqlSourceOperations :
             is And -> conj.flatMap { it.bindings() }
             is Or -> disj.flatMap { it.bindings() }
             is WhereClauseLeafNode -> {
+                log.info { "*** $column ${column.type} $bindingValue" }
                 val type = column.type as LosslessJdbcFieldType<*, *>
-                listOf(SelectQuery.Binding(bindingValue, type))
+                val v = stateValueToJsonNode(column.type, bindingValue.textValue())
```

Comment:
@xiaohansong is this the direction?
I'm not sure about this since we take the textual value and convert it to whatever type we need.
That is, we're not doing `bindingValue.intValue()`

Reply:
no - i think this needs to be as it is and we should change from the place where we call this operation by feeding it correct bindingValue. because this whereclause is a general place and not necessarily only called by parsing state value